portal/apps/org.wso2.carbon.iam.admin-portal.app/src/main/public/js/list.js

```diff
@@ -0,0 +1,343 @@
+                unlocked = true;
+            } else if ($(this.node()).hasClass(DISABLED_STATE)) {
+                disabled = true;
+            }
+        });
+        $('.bulk-element-delete').attr('disabled', false);
+        if (locked) {
+            $('.bulk-element-activate').attr('disabled', false);
+            $('.bulk-element-disable').attr('disabled', false);
+            $('.bulk-element-lock').attr('disabled', true);
+        }
+        if (unlocked) {
+            $('.bulk-element-activate').attr('disabled', true);
+            $('.bulk-element-disable').attr('disabled', false);
+            $('.bulk-element-lock').attr('disabled', false);
+        }
+        if (disabled) {
+            $('.bulk-element-activate').attr('disabled', false);
+            $('.bulk-element-disable').attr('disabled', true);
+            $('.bulk-element-lock').attr('disabled', true);
+            $('.bulk-element-delete').attr('disabled', true);
+        }
+
+        $('#users-sample').addClass("table-selectable");
+        $(button).html('Cancel');
+        $(sAllBtn).closest('li').siblings('.deselect-all-btn').show();
+        $('#filter-btn').prop('disabled', true);
+
+    }
+
+    if(selectedClaim) {
+        var i;
+        $($("#claimSelector").children()).each(
+                function(){
+                      if($(this).attr('value') == selectedClaim)
+                            i = $(this).index();
+                }
+          )
+        $('#claimSelector')[0].selectedIndex = i;
+        $('#claim-uri').val(selectedClaim)
+    }
+
+    if(selectedDomain) {
+        var j;
+        $($("#domainSelector").children()).each(
+                function(){
+                      if($(this).attr('value') == selectedDomain)
+                            j = $(this).index();
+                }
+          )
+        $('#domainSelector')[0].selectedIndex = i;
+        $('#domain-name').val(selectedDomain);
+    }
+
+} );
+
+$(document).on("click", ".open-more-modal", function () {
+    var username = $(this).data('id');
+    var modalHeading = $("#groupModalLabel").html() + ' ' + username;
+    $("#groupModalLabel").html(modalHeading);
+
+    var listString = '';
+    for (i = 0; i < moreGroups.length; i++) {
+        listString = listString + '<li>' + moreGroups[i] + '</li>';
+    }
+    listString = '<ul>' + listString + '</ul>';
+    $('.modal-body').html(listString);
+});
+
+buildColumnHeadings = function(columns) {
+    var arrayString = '{ "data": null }';
+    var columnCount = Object.keys(columns).length - 1;
+    var groupIndex;
+    var roleIndex;
+    for (var i = 0; i < columnCount; i ++) {
+        arrayString = arrayString + ',{"data": "' + columns[i] + '"}';
+        if (columns[i] === "Groups") {
+            groupIndex = i + 1;
+        } else if (columns[i] === "Roles") {
+            roleIndex = i + 1;
+        }
+    }
+    var array = JSON.parse('[' + arrayString + ']');
```

Comment:
revisit the logic to construct array from JSON objects without String manipulation